fileteasend/app.py

```diff
@@ -0,0 +1,169 @@
+#! /usr/bin/env python
+# -*- coding: utf-8 -*-
+# vim:fenc=utf-8
+#
+# Distributed under terms of the GNU GPLv3 license.
+
+from __future__ import print_function
+try:
+    from urllib.parse import urljoin, urlparse
+except ImportError:
+    from urlparse import urljoin, urlparse
+import argparse
+import json
+import logging
+import os
+import signal
+import sys
+
+import magic
+import requests
+
+class FileTeaClient(object):
+    """Base client class for FileTea access"""
+    def __init__(self, url=None):
+        logger = logging.getLogger('filetea.FileTeaClient')
+        # Set URL to default if not URL is passed
+        self.url = 'https://filetea.me/' if url == None else url
+        logger.debug('Using FileTea server {}'.format(self.url))
+        self.session = requests.session()
+
+    def get_peer_id(self):
+        """
+        Perform handshake with endpoint to get a usable peer-id
+
+        :return: peer-id
+        :rtype: string
+        """
+        logger = logging.getLogger('filetea.FileTeaClient.get_peer_id')
+        handshake_headers = {'Content-Type': 'text/plain'}
+        handshake_text = '{"mechanisms":["websocket","long-polling"],"url":"/transport/"}'
+        url = urljoin(self.url, 'transport/handshake')
+        handshake = self.session.post(url, data=handshake_text, headers=handshake_headers)
+        logger.debug(handshake.headers)
+        logger.debug(handshake.content)
+        return handshake.json()['peer-id']
+
+    def register_file(self, peer_id, file_to_send):
+        """
+        Register file with endpoint and returns url to use to download the file
+
+        :return: url
+        :rtype: string
+        """
+        logger = logging.getLogger('filetea.FileTeaClient.register_file')
+        file_name = os.path.basename(file_to_send)
+        file_size = os.path.getsize(file_to_send)
+        mime_type = magic.from_file(file_to_send, mime=True)
+        send_headers = {'Content-Type': 'text/plain'}
+        send_url = urljoin(self.url, 'transport/lp/send')
+        send_data = {
+                "method": "addFileSources",
+                "params": [[
+                    file_name,
+                    mime_type,
+                    file_size
+                          ]],
+                "id": "1"
+                    }
```

Comment:
Style nitpick: I would prefer this either at the end of the previous line or aligned at character 9 with the `s` below.